### PR TITLE
feat: add draft release PR reminder feature

### PR DESCRIPTION
**Pull request by joshjohanning**
**New PR Reminder Feature:**

* Added a new input option, `draft_release_pr_reminder`, to `action.yml` and documented it in `README.md`. This option controls whether a reminder comment is posted on the merged PR when a draft release is created. The default is `true`. [[1]](diffhunk://#diff-1243c5424efaaa19bd8e813c5e6f6da46316e63761421b3e5f5c8ced9a36e6b6R39-R42) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R34) [[3]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R76)
* Implemented logic in `src/index.js` to find the most recently merged PR associated with the current commit and post a detailed reminder comment if a draft release is created and the option is enabled. Handles cases where no PR is found or the API call fails gracefully. [[1]](diffhunk://#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R179) [[2]](diffhunk://#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556L358-R359) [[3]](diffhunk://#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R368-R411)

**Documentation Updates:**

* Updated `README.md` to describe the new input and show its usage in workflow examples. [[1]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R34) [[2]](diffhunk://#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R76)

**Testing Enhancements:**

* Expanded `__tests__/index.test.js` to mock the new GitHub API calls, verify correct comment posting behavior, and cover edge cases (e.g., no associated PR, multiple PRs, API failures, and toggling the option). [[1]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044L24-R25) [[2]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044L35-R37) [[3]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044R49-R51) [[4]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044L101-R107) [[5]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044L128-R138) [[6]](diffhunk://#diff-a49086d96004f7e5e2067ce3210c63af15d68f09d533a34ad9c8a37032a59044R959-R1150)

**Version Bump:**

* Updated `package.json` version to `2.3.0` to reflect the new feature.

### CODE REVIEW

File: __tests__/index.test.js [hunk 1]
```diff
@@ -946,4 +956,196 @@ describe('Publish GitHub Action', () => {
...
+
+      mockOctokit.rest.repos.listPullRequestsAssociatedWithCommit.mockRejectedValue(new Error('API Error'));
+
+      await run();
+
+      // Should warn but not fail
+      expect(mockCore.warning).toHaveBeenCalledWith('Could not post PR comment: API Error');
+      expect(mockCore.info).toHaveBeenCalledWith('✅ Action completed successfully!');
+    });
+  });
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply __tests__/index.test.js:1181]: actually we only care about merged PRs

File: src/index.js [hunk 2]
```diff
@@ -364,6 +365,50 @@ export async function run() {
...
+          repo: context.repo.repo,
+          commit_sha: commitShaForPr
+        });
+
+        if (prs.length > 0) {
+          // Get the most recently merged PR by sorting by merged_at date descending
+          const mergedPrs = prs
+            .filter(pr => pr.merged_at)
+            .sort((a, b) => new Date(b.merged_at) - new Date(a.merged_at));
+          const mergedPr = mergedPrs[0] || prs[0];
```
[the remark this answers is not in the document]
Comment (joshjohanning) [reply src/index.js]: actually we only care about merged PRs
